common/src/main/java/io/pravega/common/util/CertificateUtils.java

```diff
@@ -0,0 +1,110 @@
+            return extractCerts(is);
+        }
+    }
+
+    @VisibleForTesting
+    static X509Certificate[] extractCerts(@NonNull InputStream certificateInputStream) throws CertificateException {
+        Collection<? extends Certificate> certificates = CertificateFactory.getInstance("X.509")
+                .generateCertificates(certificateInputStream);
+        final X509Certificate[] result = new X509Certificate[certificates.size()];
+
+        int i = 0;
+        for (Certificate cert: certificates) {
+            result[i++] = (X509Certificate) cert;
+        }
+        return result;
+    }
+
+    public static String toString(X509Certificate[] certificateChain) {
+        StringBuilder builder = new StringBuilder();
+        for (int i = 0; i < certificateChain.length; i++) {
+            X509Certificate certificate = certificateChain[i];
+            builder.append(" -> {");
+            builder.append("type=[" + certificate.getType() + "], ");
+            builder.append("subject=[" + certificate.getSubjectX500Principal() + "], ");
+            builder.append("issuer=[" +
+                    ((certificate.getIssuerDN() != null ? certificate.getIssuerDN().getName() : "None")) + "]");
+            builder.append("}");
+        }
+        return builder.toString();
+    }
+
+    /**
+     * Creates a truststore with certificates from PEM-encoded (ASCII) or DER-encoded (binary) file at
+     * the specified {@code certFilePath}.
+     *
+     * @param certFilePath the path of the file containing a certificate chain.
+     * @return a truststore loaded with the certificate chain present in the specified {@code certFilePath}
+     * @throws CertificateException if parsing of the certificate fails
+     * @throws IOException if a file in not found at the specified {@code certFilePath}, or some other IO error occurs
+     * @throws KeyStoreException if creating an instance of the truststore fails for some reason.
+     * @throws NoSuchAlgorithmException if the algorithm used for verifying rhe truststore's integrity is missing
+     * @throws NullPointerException if {@code certFilePath} is null
+     * @throws IllegalArgumentException if {@code certFilePath} is empty
+     */
+    public static KeyStore createTrustStore(String certFilePath)
+            throws CertificateException, IOException, KeyStoreException, NoSuchAlgorithmException {
+        Exceptions.checkNotNullOrEmpty(certFilePath, "certFilePath");
+        return createTrustStore(extractCerts(certFilePath));
+    }
+
+    @VisibleForTesting
+    static KeyStore createTrustStore(X509Certificate[] certificateChain)
+            throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException {
+
+        final KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        trustStore.load(null, null);
+
+        int i = 0;
+        for (X509Certificate cert: certificateChain) {
+            String alias = i++ + "";
```

Comment:
Redundant + "" , perhaps we can use String.valueOf()

Reply:
Sure, have modified the code and pushed the change.

Reply:
Thanks @ravisharda This pattern is also present @ `io.pravega.common.util.ConfigurationOptionsExtractor#extractInt`

Reply:
Sure, thanks for spotting it. Fixed that one too.